src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -388,10 +393,22 @@ def _evaluate(
     _validate_columns(input_data_df, evaluators, target, evaluator_config)
 
     # Target Run
-    pf_client = PFClient(
-        config={"trace.destination": trace_destination} if trace_destination else None,
-        user_agent=USER_AGENT,
-    )
+    if trace_destination:
+        try:
+            pf_client = PFClient(
+                config={"trace.destination": trace_destination} if trace_destination else None,
+                user_agent=USER_AGENT,
+            )
+        except MissingAzurePackage:

```

Comment:
I suggest adding a variable skip_remote_tracking (based on the MissingAzurePackage exception) and passing it to _log_metrics_and_instance_results() to control the behavior.

Reply:
Another option is to fail the call immediately if PF Azure is NOT installed. The benefit of this approach is that it saves customers time and money, especially if they have large datasets and only want to upload the results to the cloud. Personally, I would vote for this option. Otherwise, they have to re-run the evaluation after they see the warning.

Reply:
I have set tracking URI to None in this case, so we will not try to log.

Reply:
Set it to None will end up with this error "Unable to log traces as trace destination was not defined.", which might cause confusion to customer. Why not just fail the evaluate API call?

Reply:
I have removed the code: PFClient raises the exception with comprehensive error when pf-azue is not installed.